test/test_randomness.py

```diff
@@ -96,6 +96,22 @@ def test_kaiming_normal(self):
     for shape in [(128, 64, 3, 3), (20, 24)]:
       self.assertTrue(equal_distribution(Tensor.kaiming_normal, lambda x: torch.nn.init.kaiming_normal_(torch.empty(x)), shape=shape))
 
+  def test_multinomial(self):
+    def _check_with_torch(p, num_samples, replacement):
+      tiny_res = Tensor.multinomial(Tensor(p), num_samples, replacement=replacement)
+      torch_res = torch.multinomial(torch.tensor(p), num_samples, replacement=replacement)
+      self.assertEqual(tiny_res.shape, torch_res.shape)
+      if torch_res.ndim == 1:
+        tiny_res = tiny_res.unsqueeze(0)
+        torch_res = torch_res.unsqueeze(0)
+      for i in range(torch_res.shape[0]):
+        self.assertTrue(equal_distribution(lambda *_: tiny_res[i], lambda _: torch_res[i]))
+    _check_with_torch(p=[0.231, 0., 1., 0.5], num_samples=1000, replacement=True)
+    _check_with_torch(p=[[0.453, 0., 1., 0.81], [0.1, 0.8, 0., 0.1]], num_samples=1000, replacement=True)
+    # without-replacement isn't supported, unless taking only one sample
+    self.assertRaises(AssertionError, lambda: Tensor.multinomial(Tensor([0.5, 0.5]), num_samples=100, replacement=False))
+    _check_with_torch(p=[[0.5, 0.5], [0.99, 0.01]], num_samples=1, replacement=False)
```

Comment:
Umm, doesn't this just fail 1% of the time?

Reply:
It never fails actually, because for `num_samples=1` the ks test isn't reliable. That line was just checking that no error was raised when passing `num_samples=1, replacement=False`.

I've just changed the test code to make it more reliable by accumulating the results of repeated experiments.

Regarding the reliability of `equal_distribution` in general, I wrote a counterexample to test that the samples are considered from different distributions when they should. I ran it 1000 times and it always passed. I don't know if we should keep it in the code since the other tests in the file have no such thing.

Reply:
Counterexample is great! Confirms the tests are actually testing something